packages/shield-controller/src/backend.ts

```diff
@@ -193,15 +210,26 @@ export class ShieldRemoteBackend implements ShieldBackend {
       coverageId,
     };
 
+    const abortController = this.#assignNewAbortController(configs.requestId);
+
     const timeout = configs?.timeout ?? this.#getCoverageResultTimeout;
     const pollInterval =
       configs?.pollInterval ?? this.#getCoverageResultPollInterval;
 
     const headers = await this.#createHeaders();
     return await new Promise((resolve, reject) => {
       let timeoutReached = false;
+
+      const abortHandler = () => {
+        timeoutReached = true;
+        this.#removeAbortHandler(configs.requestId, abortHandler);
```

Comment:
Unrelated to here, but could we change `configs` to `config`? Configuration is usually used in singular. Alternatively, you could use `opts` or `options`.